File: programmers/js_src/level01/42862.js
// 체육복

function solution(n, lost, reserve) {
  var answer = 0;

  lost.sort((a, b) => a - b);
  reserve.sort((a, b) => a - b);

  let _reserve = reserve.filter(num => !lost.includes(num));
  let _lost = lost.filter(num => !reserve.includes(num));

  answer = n - _lost.length;

  for (let num of _reserve) {
    if (_lost.includes(num - 1)) {
      _lost.splice(_lost.indexOf(num - 1), 1)
      answer++;
    } else if (_lost.includes(num + 1)) {
      _lost.splice(_lost.indexOf(num + 1), 1)
      answer++;
    }
  }

  return answer;
}

File: programmers/java/level01/42862.java
// 체육복

import java.util.Arrays;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int[] result = new int[n + 1];
        Arrays.fill(result, 1);
        
        for (int i = 0; i < lost.length; i++) {
            result[lost[i]] -= 1;
        }
        
        for (int i = 0; i < reserve.length; i++) {
            result[reserve[i]] += 1;
        }
        
        for (int i = 1; i <= n; i++) {
            if (result[i] == 0) {
                if (result[i - 1] == 2) {
                    result[i] = 1;
                    result[i - 1] = 1;
                } else if (i < n && result[i + 1] == 2) {
                    result[i] = 1;
                    result[i + 1] = 1;
                }
            }
        }
        
        return (int) Arrays.stream(result).filter(v -> 0 < v).count() - 1;
    }
}